implicits/src-3/upickle/implicits/CaseClassReader.scala

```diff
@@ -64,7 +64,13 @@ trait CaseClassReaderPiece extends MacrosCommon:
           if (!missingKeys.isEmpty) {
             throw new upickle.core.Abort("missing keys in dictionary: " + missingKeys.mkString(", "))
           }
-          m.fromProduct(ArrayProduct(values.toArray))
+
+          val valuesArray = values.toArray
+          m.fromProduct(new Product {
+            def canEqual(that: Any): Boolean = true
+            def productArity: Int = valuesArray.length
+            def productElement(i: Int): Any = valuesArray(i)
+          })
```

Comment:
It does look like ArrayProduct was useful after all.

Reply:
I or we never said it wasn't useful at all. Just that its usefulness was too limited to deserve a place in the standard library, compared to all the other things that are not there and that could be useful too (even just for deriving).